core/src/SYCL/Kokkos_SYCL.hpp

```diff
@@ -159,8 +159,14 @@ std::vector<SYCL> partition_space(const SYCL& sycl_space, Args...) {
   std::vector<SYCL> instances;
   instances.reserve(sizeof...(Args));
   for (unsigned int i = 0; i < sizeof...(Args); ++i)
-    instances.emplace_back(
-        sycl::queue(context, device, sycl::property::queue::in_order()));
+    instances.emplace_back(sycl::queue(
+        context, device,
+        {sycl::property::queue::in_order()
```

Comment:
Was it intentional that the use of "in order" queue was not guarded here?

Reply:
Related pull requests are https://github.com/kokkos/kokkos/pull/6189 and https://github.com/kokkos/kokkos/pull/6246. The latter former made the default execution space instances in order and the latter dealt with interoperability (without the guard). Thus, it's at least consistent with the interoperability test but I could see that would also be consistent to guard it here.

Reply:
I don't understand your answer.  If we have a `KOKKOS_IMPL_SYCL_USE_IN_ORDER_QUEUES` macro and we use it internally to decide whether to pass the `sycl::property::queue::in_order` for the default instance, why are we unconditionally specifying `in_order` in `partition_space`?